### PR TITLE
Change CassandraType enum to class

### PR DESCRIPTION
**Pull request by ebyhr**
We should avoid enum for complex type mapping (e.g. UDT in Cassandra).

### CODE REVIEW

File: plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraType.java [hunk 1]
```diff
@@ -463,48 +458,48 @@ public boolean isSupportedPartitionKey()
...
+        if (!isSupported(dataType.getName())) {
             return false;
         }
 
         return dataType.getTypeArguments().stream()
                 .allMatch(CassandraType::isFullySupported);
     }
 
-    public static CassandraType toCassandraType(Type type, ProtocolVersion protocolVersion)
+    public static DataType.Name toCassandraType(Type type, ProtocolVersion protocolVersion)
```
Comment (losipiuk) [inline plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraType.java:469]: nit: change method name to `toCassandraTypeName`

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraColumnHandle.java [hunk 2]
```diff
@@ -44,7 +45,7 @@ public void testRoundTrip2()
         CassandraColumnHandle expected = new CassandraColumnHandle(
                 "name2",
                 1,
-                CassandraType.MAP,
+                DataType.Name.MAP,
```
Comment (losipiuk) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraColumnHandle.java:48]: It feels to me that we should keep `CassandraType` here. `DataType.Name` does not pass all the needed information if we want to support parametrized types.

**Review by losipiuk — COMMENTED** [review]
Thanks! Good direction.

This is mostly in line with changes I have shelved in https://github.com/trinodb/trino/pull/8567/ (first commit). The PR requires so final work but I did not have time recently to finish that up.

What are you pursuing eventually? I guess this is not just a refactor for its own sake.

**Comment by findepi** [comment]
We need this to properly support collections in Cassandra.

**Comment by ebyhr** [comment]
Yes, the main motivation is changing collection type mapping from VARCHAR. 
Let's continue in @losipiuk's PR.

**Pull request closed without merging**